### PR TITLE
Post model update

### PR DESCRIPTION
**Pull request by Brett-BI**
models.py
- Changed Post to Article; updated verbiage accordingly
- Updated property names
- Added `__tablename__`
- Added `@classmethods`
    - `insert_article`
    - `get_article`
    - `get_random_article`
- Created new migration file

reader_routes
- Updated references to models.Article (was models.Post)
- Added calls to appropriate `@classmethod`s

templates/filters
- Updated references to new Article model in some instances
- Changed `display_date` filter to return "Today" for any article posted <= 1 day previously

general
- Added pyodbc to requirements.txt
- Added psyopg2 to requirements.txt

### CODE REVIEW

File: app/reader/filters.py [hunk 1]
```diff
@@ -5,8 +5,9 @@
 @reader_bp.app_template_filter('display_date')
 def display_date(dt):
     diff = datetime.now() - dt
```
Comment (maxmarzolf) [inline app/reader/filters.py:7]: I've always been a fan of variable names that aren't abbreviated, would have gone with something like time_elapsed = datetime.now() - dt

File: app/reader/filters.py [hunk 2]
```diff
@@ -5,8 +5,9 @@
...
+    print(diff)
 
-    if diff.days == 1:
+    if diff.days <= 1:
         return "Today"
     else:
         #return f"{dt.month} {dt.day}, {dt.year}"
```
Comment (maxmarzolf) [inline app/reader/filters.py:13]: I've heard the community rave about fstrings, why did you change your mind?

File: app/reader/reader_routes.py [hunk 3]
```diff
@@ -8,16 +8,20 @@
 
 @reader_bp.route("/")
 def reader_home():
-    posts_from_db = models.Post.query.order_by(desc(models.Post.post_date)).limit(4)
+    #posts_from_db = models.Post.query.order_by(desc(models.Post.post_date)).limit(4)
+    posts_from_db = models.Article.get_random_articles(4)
```
Comment (maxmarzolf) [inline app/reader/reader_routes.py:13]: hahahahaha, I like this, I interpret it as getting random articles instead of the latest 4 articles? Am I correct about that?

File: app/reader/filters.py (hunk 1 above)
Comment (Brett-BI) [reply app/reader/filters.py:7]: Agreed. We could probably just look to remove the abbreviations everywhere for clarity.

File: app/reader/filters.py (hunk 2 above)
Comment (Brett-BI) [reply app/reader/filters.py:13]: I just realized now that it should not be commented. I wanted to display "today" for recent articles but the actual date for all others. My bad.

File: app/reader/filters.py (hunk 1 above)
Comment (maxmarzolf) [reply app/reader/filters.py:7]: Sounds good, dude this is awesome, GitHub rocks.

File: app/reader/reader_routes.py (hunk 3 above)
Comment (Brett-BI) [reply app/reader/reader_routes.py:13]: You're correct. It's a good example of that @classmethod thing we were talking about. It's so much cleaner to put a method on the Article class that gets a number of random articles with a parameter for the number.

While I also like the idea of getting random articles more, we could easily add a get_last_articles_by_date method to get them in order. Possibilities are endless and I can see both being useful.

File: app/reader/filters.py (hunk 2 above)
Comment (maxmarzolf) [reply app/reader/filters.py:13]: cool cool, this isn't related to this comment, but I wonder what the difference between the "Start a Review" and conversation is.

Comment (Brett-BI) [reply app/reader/filters.py:13]: Not sure. It would probably contribute to the Code Review stats though.

File: app/reader/reader_routes.py (hunk 3 above)
Comment (maxmarzolf) [reply app/reader/reader_routes.py:13]: Interesting, so the get_random_articles is a @classmethod in the Article object, very cool. Now what was the initial and less clean way of doing it?

Comment (Brett-BI) [reply app/reader/reader_routes.py:13]: The initial way was executing a query. The line that was removed was the original way. There are two places that random articles are grabbed so it didn't make sense to keep writing that query. 

I also wasn't a fan of dumping that SQLAlchemy stuff in a route. It just looks more organized to me if that business logic is placed in the models as much as possible. I don't have enough experience to say that we can do the same everywhere but I'd like to try. 

Eventually, I'd like to look into a way to have the models return stuff that is immediately usable by the front end. You can see there's still some logic on some of the routes to break apart the SQLAlchemy query objects into a regular dict that the templates can use. I want to move that into the models as much as possible but I'm not sure what options there are, if any. 

I guess overall, I'm just looking to have the routes contain a minimal amount of code. The heavy lifting can be done in the models. It's definitely something we can discuss though. I think it's largely a stylistic choice.
